lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -24,25 +24,27 @@
 using Org.Apache.REEF.IO.PartitionedData.Random.Parameters;
 using Org.Apache.REEF.IO.TempFileCreation;
 using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Attributes;
 using Org.Apache.REEF.Utilities.Diagnostics;
 using Org.Apache.REEF.Utilities.Logging;
 
 namespace Org.Apache.REEF.IO.PartitionedData.FileSystem
 {
+    [ThreadSafe]
     internal sealed class FileSystemInputPartition<T> : IInputPartition<T>, IDisposable
     {
         private static readonly Logger Logger = Logger.GetLogger(typeof(FileSystemInputPartition<T>));
 
         private readonly string _id;
         private readonly IFileSystem _fileSystem;
         private readonly IFileDeSerializer<T> _fileSerializer;
-        private readonly ISet<string> _filePaths;
-        private bool _isInitialized;
-        private readonly bool _copyToLocal;
         private readonly object _lock = new object();
         private readonly ITempFileCreator _tempFileCreator;
-        private readonly ISet<string> _localFiles = new HashSet<string>();
+        private readonly ISet<string> _filePaths;
```

Comment:
I think we should be explicit on what this is for. If this is only for remote files, we should make it so. For testing, we should fake the `IFileSerializer` to fetch from local instead of using an `if` branch the way we do right now that obfuscates the intention of the code.